tests/zquantum/qaoa/estimators/gibbs_estimator_test.py

```diff
@@ -0,0 +1,82 @@
+import pytest
+from openfermion import QubitOperator, IsingOperator
+
+from zquantum.core.interfaces.mock_objects import (
+    MockQuantumBackend,
+)
+from zquantum.core.circuits import Circuit, X, H
+from zquantum.core.measurement import Measurements
+from zquantum.qaoa.estimators import GibbsEstimator
+
+from zquantum.core.interfaces.estimation import EstimationTask
+import numpy as np
+
+
+class TestGibbsEstimator:
+    @pytest.fixture(params=[1.0, 0.8, 0.5, 0.2])
+    def estimator(self, request):
+        return GibbsEstimator(alpha=request.param)
+
+    @pytest.fixture()
+    def circuit(self):
+        return Circuit([X(0)])
+
+    @pytest.fixture()
+    def operator(self):
+        return IsingOperator("Z0")
+
+    @pytest.fixture()
+    def estimation_tasks(self, operator, circuit):
+        return [EstimationTask(operator, circuit, 10)]
+
+    @pytest.fixture()
+    def backend(self, request):
+        backend = MockQuantumBackend()
+
+        def custom_run_circuit_and_measure(circuit, n_samples):
+            bitstrings = [("0"), ("1")]
+            return Measurements(bitstrings)
+
+        backend.run_circuit_and_measure = custom_run_circuit_and_measure
+        return backend
+
+    def test_raises_exception_if_operator_is_not_ising(
+        self, estimator, backend, circuit
+    ):
+        # Given
+        estimation_tasks = [EstimationTask(QubitOperator("X0"), circuit, 10)]
+        with pytest.raises(TypeError):
+            estimator(
+                backend=backend,
+                estimation_tasks=estimation_tasks,
+            )
+
+    def test_gibbs_estimator_raises_exception_if_alpha_less_than_0(
```

Comment:
It's good to add test cases for boundary conditions as well – in this case for `alpha=0`.